src/utils/dataset_utils.py

```diff
@@ -526,39 +497,106 @@ def parse_file(self, fp, file_name):
+    """Simple parsing of TSV file drawn from Elasticsearch"""
+    
+    def __init__(self, path, encoding="utf-8", max_docs=float('inf'), verbose=False, clean_docs=False,
+                 content_fields=['extracted_text'], term='', max_doc_length=0):
+        super().__init__(path, encoding=encoding, max_docs=max_docs)
+        self.verbose = verbose
+        self.clean_docs = clean_docs
+        self.content_fields=content_fields
+        self.term=term
+        self.max_doc_length=max_doc_length
+        
+    def parse_file(self, fp, file_name):
+        with codecs.open(fp, encoding=self.encoding) as tsv:
+            for ind, line in enumerate(tsv):
+                if ind == 0:
+                    continue
+                try:
+                    # Loading data -- ignore malformatted entries!
+                    # TODO: Make these fields dynamic/drawn from header? Or make field names an option?
+                    (doc_id, uuid, memex_id, memex_content_type, crawl_data, memex_crawler, memex_doc_type, memex_extracted_metadata, memex_extracted_text, memex_extractions, memex_raw_content, memex_team, memex_timestamp, memex_type, memex_url, memex_version, domain, content_type, url, content, extractions) = line.split('\t')
+                except:
+                    print('Malformatted Line!')
+                    continue
+
                 # Cleaning documents if specified
                 if self.clean_docs:
-                    doc_text = doc_text.replace('\n',' ').replace('\t',' ').replace('<br>', ' ').replace('\\\\n', ' ')
-                    # Eliminating extra space
-                    doc_text = " ".join(doc_text.split())
+                    content = clean_input(content)
+                    memex_raw_content = get_posting_html(memex_raw_content, self.term)
+                    memex_url = parse_url(memex_url)
+
+                doc_text = ""
+                field_names = {'extracted_text': content, 'raw_content': memex_raw_content, 'url': memex_url, }
+                for field in self.content_fields:
+                    doc_text += field_names[field] + " "
 
+                doc_text = doc_text.strip()
+                doc_name = doc_id
+                source = content_type
+                extractions = extractions
+                
+                # Short documents are usually parsing errors...
+                if len(doc_text) < 10:
+                    if self.verbose:
+                        print('Short Doc!')
+                    continue
+       
+                # long documents sometimes causing parsing to stall
+                if self.max_doc_length and len(doc_text) > self.max_doc_length:
+                    if self.verbose:
+                        print('Long document')
+                    continue
+#                    doc_text = doc_text[-self.max_doc_length:]
+ 
+                # Setting stable id
+                stable_id = self.get_stable_id(doc_name)
+                
```

Comment:
Move `ParallelESTSVPreprocessor` after `ESTSVPreprocessor` to avoid giant diff :) .

Reply:
Done